### DISCUSSION TITLE
Add middleware system into message processor

### DISCUSSION THREAD

**Discussion by dohaicuong** [viet-aus-it/vait-discord-bot#59] [Ideas]
We can have a middleware system to implement some extra security, limit rate, log, auth logic into message processor before it comes into commands code.

**Comment by samhwang** [comment]
MIddleware ideas:

1/ isBot: Detect if the message was sent from a bot

2/ isBlank: Detect if the message was blank

3/ isAdmin: Detect if the message was sent by an admin

4/ Another middleware to handle sending the result message, so that we don't have to run a try/catch every time.

**Reply by daniel-nguyen24** [reply]
Adding to this:

5/ isValidUser: Check whether users have agreed to the rules in #rules and picked up some roles in #get-your-role
